File: chapter 07/recipe 7.4/DniNetVariants/DniNetVariantsObj.cs
using System;
using System.Runtime.InteropServices;

namespace DniNetVariants
{
	public interface IVariants
	{
		string UseVariant(Object inParam, ref Object outParam);
	}

	[ClassInterface(ClassInterfaceType.None)]
	public class DniNetVariantsObj : IVariants
	{
		public string UseVariant(object inParam, ref object outParam)
		{
			//copy the input param to the output
			outParam = inParam;

			//return the type description
			return inParam.GetType().Name;
		}
	}
}

File: chapter 07/recipe 7.4/DniNetVariantsVB/DniNetVariantsObj.vb
Imports System.Runtime.InteropServices

Public Interface IVariants
    Function UseVariant(ByVal inParam As Object, _
        ByRef outParam As Object) As String
End Interface

<ClassInterface(ClassInterfaceType.None)> _
Public Class DniNetVariantsObj
    Implements IVariants

    Public Function UseVariant(ByVal inParam As Object, _
        ByRef outParam As Object) As String _
            Implements IVariants.UseVariant
        'copy the input param to the output
        outParam = inParam

        'return the type description
        Return inParam.GetType().Name
    End Function
End Class

Public Interface ITestSyntax
    Function MyObjectMethod(ByVal param As Object) As Object
End Interface